### PR TITLE
Export s6

### PR DESCRIPTION
**Pull request by hectorcorrea**
*\* DO NOT MERGE **

This is a first pass at exporting ScholarSphere's GenericFiles and Collections stored under a Sufia 6 implementation to a set of JSON files that can be used to import the data into a Sufia 7 implementation. 

To export Generic Files the idea is to run two rake tasks. One to export the IDs of all the files to export and another to actually export them to JSON files (one per Generic File). I suspect we could combine these tasks but for a final release. 

```
rake gf_ids > gf_ids.txt
rake export_gf[gf_ids.txt]
```

Similar rake tasks can be used to export Collections.

You can see a sample of the JSON files generated when exporting a file and a collection here: https://gist.github.com/hectorcorrea/619c13fa540341be14cb01c762d9f2cc

The JSON file generated for Generic Files only includes the metadata (including metadata for versions and permissions) but not the actual binary. The idea is to fetch the binary (for the current and previous versions) directly from Fedora 4 at the time the data is imported to a Sufia 7 application.

### CODE REVIEW

File: lib/export/generic_file_metadata_export.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      @date_modified = gf.date_modified
+      @subject = gf.subject
+      @language = gf.language
+      @identifier = gf.identifier
+      @based_near = gf.based_near
+      @related_url = gf.related_url
+      @bibliographic_citation = gf.bibliographic_citation
+      @source = gf.source
+      @batch_id = gf.batch.id if gf.batch
+      @visibility = gf.visibility
```
Comment (awead) [inline lib/export/generic_file_metadata_export.rb:45]: use delegate_to for all of these :smile:

File: lib/export/versions_from_graph.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    end
+
+    def self.find_uris(graph)
+      uris = []
+      graph.each do |triple|
+        if triple.predicate.to_s == "http://fedora.info/definitions/v4/repository#hasVersion"
+          uris << triple.object.to_s
+        end
+      end
+      uris
```
Comment (awead) [inline lib/export/versions_from_graph.rb:35]: I think you can do something like:

``` ruby
graph.query(predicate: ::RDF::Vocab::Fcrepo4.hasVersion).map(&:to_s)
```

File: lib/export/versions_from_graph.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+module Export
+  class VersionsFromGraph
+    attr_accessor :versions
```
Comment (awead) [inline lib/export/versions_from_graph.rb:4]: is `ActiveFedora::VersionsGraph` any help here? Not sure, just wanted to make sure you knew about it.

File: lib/export/generic_file_metadata_export.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      @date_uploaded = gf.date_uploaded
+      @date_modified = gf.date_modified
+      @subject = gf.subject
+      @language = gf.language
+      @identifier = gf.identifier
+      @based_near = gf.based_near
+      @related_url = gf.related_url
+      @bibliographic_citation = gf.bibliographic_citation
+      @source = gf.source
+      @batch_id = gf.batch.id if gf.batch
```
Comment (carolyncole) [inline lib/export/generic_file_metadata_export.rb:44]: Change this to list the related IDs from the original batch inside the Item.

File: lib/export/collection_metadata_export.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+    def initialize(coll)
+      @id = coll.id
+      @title = coll.title
+      @description = coll.description
+      @creator = coll.creator.map { |c| c }
+      @members = coll.members.map(&:id)
+    end
+
+    def to_json(pretty = false)
```
Comment (carolyncole) [inline lib/export/collection_metadata_export.rb:17]: Overrode so that it is easier to debug.  This is not needed for running.

**Pull request closed without merging**